lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java

```diff
@@ -0,0 +1,132 @@
+    defaultWorkbasket =
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStore(workbasketService);
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+
+    taskWithComments =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassification.asSummary())
+            .workbasketSummary(defaultWorkbasket.asSummary())
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(taskWithComments.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+    comment2 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(taskWithComments.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_CreateTaskComment_For_TaskThatAlreadyHasComments() throws Exception {
+    TaskComment taskCommentToCreate = taskService.newTaskComment(taskWithComments.getId());
+    taskCommentToCreate.setTextField("Some text");
+    taskService.createTaskComment(taskCommentToCreate);
+
+    List<TaskComment> taskCommentsAfterInsert =
+        taskService.getTaskComments(taskWithComments.getId());
+    assertThat(taskCommentsAfterInsert)
+        .containsExactlyInAnyOrder(comment1, comment2, taskCommentToCreate);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_FailToCreateTaskComment_When_UserHasNoWorkbasketPermission() {
+    TaskComment taskCommentToCreate = taskService.newTaskComment(taskWithComments.getId());
+    taskCommentToCreate.setTextField("Some text");
+
+    assertThatThrownBy(() -> taskService.createTaskComment(taskCommentToCreate))
+        .isInstanceOf(TaskanaException.class)
+        .hasMessageContaining("The current user 'user-1-2' has no '[READ]' permission");
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_CreateTaskComment_When_CopyingAnotherComment() throws Exception {
+    TaskComment taskCommentToCreate = comment1.copy();
+    int sizeBeforeInsertingComment = taskService.getTaskComments(taskWithComments.getId()).size();
+
+    taskService.createTaskComment(taskCommentToCreate);
+
+    List<TaskComment> taskCommentsAfterInsert =
+        taskService.getTaskComments(taskWithComments.getId());
+    assertThat(taskCommentsAfterInsert)
+        .contains(comment1, taskCommentToCreate)
+        .hasSize(sizeBeforeInsertingComment + 1);
+  }
```

Comment:
I think, the assertion approach with containsExactlyInAnyOrder you used `should_CreateTaskComment_For_TaskThatAlreadyHasComments()` is better and smarter! You can reuse it.

Reply:
Ok :)